### DISCUSSION TITLE
Using DefaultAzureCredential in iot-service-client 2.0.2

### DISCUSSION THREAD

**Discussion by HarryAcker** [Azure/azure-iot-sdk-java#1569] [Q&A]
When creating the message client to send C2D should the DefaultAzureCredential work for authentication.  When I create the MessagingClient it looks like it works, the open method times out 3 or 4 times but then returns OK but a call to isOpen return false.  The exact same code works fine with a connection string.  If the token was no good I would expect an unauthorized exception when trying to open the client.  Anyone tried the DefaultAzureCredential in this scenario?  And just to verify in the same code I use DefaultAzureCredential to create a KV secret client and that works 100%.

**Comment by timtay-microsoft** [comment]
Do you have a code snippet that demonstrates this issue? I'd be happy to try it out on my end to see what is happening.

**Reply by timtay-microsoft** [reply]
@HarryAcker
